packages/wallet/src/components/dialogs/SuccessDialog.tsx

```diff
@@ -59,25 +61,28 @@ export const SuccessDialog = ({
                   </div>
                 </div>
                 <div className="mt-5 grid grid-cols-1 gap-3">
-                  <Button
-                    intent="success"
-                    aria-label="close dialog"
-                    fullWidth
-                    onClick={() => onClose()}
-                  >
-                    Close
-                  </Button>
                   {redirect && (
                     <Button
                       intent="secondary"
                       aria-label={redirectText ?? 'redirect'}
                       fullWidth
                       href={redirect ?? '/'}
-                      onClick={() => onClose()}
+                      onClick={() => {
```

Comment:
I've understood the idea of `onSuccess`. The problem is that we have an `href` and `onClick`. These will both execute when the button is clicked.